mitodl/ocw-studio
# Replace `@ts-ignore` with `@ts-expect-error`, and remove most of them

#### Pre-Flight checklist


#### What are the relevant tickets?
Toward https://github.com/mitodl/eslint-config/issues/14, updating our eslint config for typescript

#### What's this PR do?
This PR:
1. Replaces all `@ts-ignore` with `@ts-expect-error`. The latter has the advantage that if it is used unnecessarily, TS will yell at you.
2. Additionally, I made some fairly minor code changes to reduce the number of `@ts-expect-error`s needed. For the most part, these were in test files, things like
    - using `const thing = jest.mocked(originalThing)` which does nothing but change the object's type to a jest mock (with signature matching `originalThing`).
    - changed a bunch of enzyme calls like `wrapper.find("StringComponentName")` to `wrapper.find(ComponentClass)`. That way TS knows the types of props, etc
    - Adding `as const` to our test case arrays (or rearranging them to objects)
3. I updated our CI's node version to 16.15, to match docker. My motivation was that CI said [Array.prototype.at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) was not a function, though I ended up switching to `lodash.last` since `at` only has 90% browser support.

    But, still seems good for CI to match docker.

**All in all:** We had 176 ignore/expect-errors. Now we have 44 expect-errors.

#### How should this be manually tested?
There should be no behavior change, but if you want, check ... that Studio still works.

#### Where should the reviewer start?

⚠️ ⚠️ ⚠️ Strongly suggest reviewing this commit-by-commit. https://github.com/mitodl/ocw-studio/pull/1520/commits/d79aa320418286a1a581d4ec08eb7fbde76afa4d is the only commit that could affect app behavior; the other commits are all about linting or tests.
